network_wrangler/scenario.py

```diff
@@ -38,627 +41,530 @@ class Scenario(object):
+            self.__dict__[d].update({project_name: _dep})
 
-        i.e. glob_search = 'road*.yml'
+    def _add_project(
+        self,
+        project_card: ProjectCard,
+        validate: bool = True,
+        filter_tags: Collection[str] = [],
+    ) -> None:
+        """Adds a single ProjectCard instances to the Scenario.
 
-        args:
-        folder: the folder location where the project cards will be
-        glob_search: https://docs.python.org/2/library/glob.html
-        """
+        Checks that a project of same name is not already in scenario.
+        If selected, will validate ProjectCard before adding.
+        If provided, will only add ProjectCard if it matches at least one filter_tags.
 
-        if not os.path.exists(folder):
-            msg = "Cannot find specified directory to add project cards: {}".format(
-                folder
-            )
-            WranglerLogger.error(msg)
-            raise ValueError(msg)
+        Resets scenario queued_projects.
 
-        if glob_search:
-            WranglerLogger.info(
-                "Adding project cards using glob search: {}".format(glob_search)
-            )
-            for file in glob.iglob(os.path.join(folder, glob_search)):
-                if not (
-                    file.endswith(".yml")
-                    or file.endswith(".yaml")
-                    or file.endswith(".wrangler")
-                    or file.endswith(".wr")
-                ):
-                    continue
-                else:
-                    self.add_project_card_from_file(file, validate=validate)
-        else:
-            for file in os.listdir(folder):
-                if not (
-                    file.endswith(".yml")
-                    or file.endswith(".yaml")
-                    or file.endswith(".wrangler")
-                    or file.endswith(".wr")
-                ):
-                    continue
-                else:
-                    self.add_project_card_from_file(
-                        os.path.join(folder, file), validate=validate
-                    )
+        Args:
+            project_card (ProjectCard): ProjectCard instance to add to scenario.
+            validate (bool, optional): If True, will validate the projectcard before
+                being adding it to the scenario. Defaults to True.
+            filter_tags (Collection[str], optional): If used, will only add the project card if
+                its tags match one or more of these filter_tags. Defaults to []
+                which means no tag-filtering will occur.
 
-    def add_project_cards_from_tags(
-        self, folder: str, tags: List[str] = [], glob_search="", validate=True
-    ):
         """
-        Adds projects cards to the scenario.
-        A folder is provided to look for project cards that have a matching tag that
-        is passed to the method.
+        project_name = project_card.project.lower()
+        filter_tags = map(str.lower, filter_tags)
```

Comment:
errors out when `filter_tags` is None

Reply:
It shouldn't ever be None - I can't see where it would be set to None?

Reply:
filter_tags shouldn't ever be `None`, it should be `[]` when not provided